src/lib/transaction_snark/transaction_snark.ml

```diff
@@ -2535,11 +2610,20 @@ module Base = struct
                Receipt.Chain_hash.Checked.if_ is_user_command ~then_:r
                  ~else_:current
              in
+             let update_not_permitted =
+               transaction_failure.fee_payer_balance_update_not_allowed
```

Comment:
This should be a hard failure, not a soft one. The fee payment should always succeed!!

Reply:
But i guess it needs to fail from the check inside the circuit